File: ruby/queue/test_queue.rb
require_relative 'queue'
require 'test/unit'

class TestLinkedList < Test::Unit::TestCase
  def setup
    @q = Queue.new(1)
    @q.enqueue(2)
    @q.enqueue(3)
  end

  def test_queue
    assert_equal(@q.peek, 1)
    assert_equal(@q.dequeue, 1)
    @q.enqueue(4)
    assert_equal(@q.dequeue, 2)
    assert_equal(@q.dequeue, 3)
    assert_equal(@q.dequeue, 4)
    @q.enqueue(5)
    assert_equal(@q.peek, 5)
  end
end

File: python/queue/test_queue.py
import unittest
from queue import *

class QueueTest(unittest.TestCase):
    def setUp(self):
        self.q = Queue(1)
        self.q.enqueue(2)
        self.q.enqueue(3)

    def test_queue(self):
        self.assertEqual(self.q.peek(), 1)
        self.assertEqual(self.q.dequeue(), 1)
        self.q.enqueue(4)
        self.assertEqual(self.q.dequeue(), 2)
        self.assertEqual(self.q.dequeue(), 3)
        self.assertEqual(self.q.dequeue(), 4)
        self.q.enqueue(5)
        self.assertEqual(self.q.peek(), 5)

if __name__ == '__main__':
    unittest.main()
